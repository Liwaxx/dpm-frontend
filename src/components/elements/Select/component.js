import React from 'react';
import TextField from '@material-ui/core/TextField';
import PropTypes from 'prop-types';

export default class Component extends React.Component {
  constructor() {
    super();
    this.state = {
      filter: '',
    };
  }

  filterOptions = [
    { label: 'Most Viewed', value: 'most' },
    { label: 'Recent Update', value: 'recent' },
  ];

  handleChange = prop => event => {
    this.setState({ [prop]: event.target.value });
  };

  _renderSelectField(options) {
    const { classes } = this.props;

    return (
      options.map(item => (
        <option
          className={classes.menu}
          key={item.value}
          value={item.value}
        >
          {item.label}
        </option>
      ))
    );
  }

  render() {
    const { classes } = this.props;
    const { filter } = this.state;
    const InputProps = {
      disableUnderline: true,
    };
    const SelectProps = {
      className: classes.select,
      native: true,
    };

    const value = filter == '' ? '' : filter;

    return (
      <TextField
        select
        className={classes.dropdown}
        InputProps={InputProps}
        onChange={this.handleChange('filter')}
        SelectProps={SelectProps}
        value={value}
      >
        <option value="" disabled>Filters</option>
        {this._renderSelectField(this.filterOptions)}
      </TextField>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object
};

