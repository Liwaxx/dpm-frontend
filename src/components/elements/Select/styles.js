const styles = {
  menu: {
    backgroundColor: '#ffffff',
    fontSize: '14px',
    padding: '0 8px',
  },
  dropdown: {
    backgroundColor: '#ffffff',
    border: '1px solid #AEB9CB',
    borderRadius: '2px',
    width: '130px',
  },
  select: {
    fontSize: '14px',
    padding: '0 8px',
  },
};

export default styles;