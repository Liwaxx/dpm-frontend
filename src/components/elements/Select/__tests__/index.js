import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../index';

describe('Select', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Select />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});