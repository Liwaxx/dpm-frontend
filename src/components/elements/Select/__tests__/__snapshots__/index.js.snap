// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div
  className="MuiFormControl-root-4 Component-dropdown-2"
>
  <div
    className="MuiInput-root-13 MuiInput-formControl-14 Component-select-3"
  >
    <div
      className="MuiSelect-root-8"
    >
      <select
        aria-invalid={false}
        className="MuiSelect-select-9 MuiInput-input-21"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        value=""
      >
        <option
          disabled={true}
          value=""
        >
          Filters
        </option>
        <option
          className="Component-menu-1"
          value="most"
        >
          Most Viewed
        </option>
        <option
          className="Component-menu-1"
          value="recent"
        >
          Recent Update
        </option>
      </select>
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-26 MuiSelect-icon-12"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
